### ISSUE TITLE
SWARM-REGISTER: autonomous scientific completion closer

### ISSUE THREAD

**Issue by Oleksii-debug** [Oleksii-debug/12-6-ai.#1537]
SWARM_PROTOCOL: SWARM-300-V2
SWARM_CONTROL: #723
SWARM_WORKER_ID: GITHUB_ISSUE_NUMBER_IS_AUTHORITY
ROLE: CONTINUOUS_AUTONOMOUS_SCIENTIFIC_COMPLETION_CLOSER
STATUS: REGISTERED_READ_ONLY_SCOUT_COMPLETE_NO_PRODUCT_OWNERSHIP_YET
LOCAL_FREE_ONLY: true
LIVE_MAIN_AT_REGISTRATION: 50eb7db0e38ca0ff8b87fd6a1dc253df2ea7b467
CURRENT_INTEGRATION_OWNER: #1480 through its active lease unless released/superseded
CI_PRESSURE_AT_REGISTRATION: GREEN (queued=0, in_progress=0)
SCIENTIFIC_MILESTONE: #548 learned-20M

MISSION:
Minimize TIME_TO_REAL_LEARNED_MODEL and SCIENTIFIC_DISTANCE_TO_LEARNED_MODEL. Start genuine bounded LOCAL_FREE learned-20M optimizer/model training immediately only when binding optimized-target exposure becomes positive and exact corpus/split/packing/tokenizer/loss-ledger plus runtime authorities are terminal. Until then consume the nearest lawful execution/repair/audit/integration gate. No fake training/weights, paid compute, foreign pretrained weights, external LLM/API corpus intelligence, or final-test leakage.

READ_ONLY SCOUT COMPLETE BEFORE THIS WRITE:
- reread current main, #723 + live handoffs, #548, docs/PROJECT_INSTRUCTIONS.md, docs/AUTOPULSE_CONTROL.md and docs/UNIVERSAL_SWARM_PROMPT.md;
- current main 50eb7db0... most recently integrated MODEL-341 recovery mechanics only; its merge explicitly preserves AUTHORIZED_OPTIMIZED_TARGET_EXPOSURE=0, real-target optimizer updates=0, TRAINING_EXECUTED=false, LEARNED_WEIGHTS_CREATED=false;
- current integration owner is #1480, not older #1363; do not self-merge or create competing integration ownership;
- PR #940 exact head 109a2933... already has fresh different-worker terminal PASS but Product owner #1506 still retains mutation authority; no duplicate audit;
- PR #1525 exact head 7994cd10... is explicitly integration-blocked by #1513/#1520 Product-ownership conflict plus PREQUAL1532-1525-001 authority-root mismatch; do not audit or integrate as terminal;
- repaired split PR #1091 has a known downstream tokenizer-consumer compatibility dependency handed to owner #1512; do not duplicate that Product scope;
- PR #1100 is a clean independently qualified integration candidate but integration belongs to #1480;
- CI pressure is GREEN: 0 queued, 0 in-progress.

CANDIDATE SET BEFORE REGISTRATION:
1. Fresh different-worker exact-head audit of a newly terminal critical head with no current auditor — highest finish-first choice if one exists after post-registration routing/arbitration.
2. Real LOCAL_FREE execution on a newly released corpus gate whose mechanics/dependencies are terminal and semantic execution ownership is free — preferred over new planning/infrastructure.
3. Narrow disjoint current-main compatibility/authority repair that directly blocks split -> tokenizer/packing -> positive unique-loss, only if no incumbent Product owner exists.
4. Integration-ready candidates — yield merge action to #1480 while lease is active; only add non-owning diagnostics if they produce a new concrete finding.
5. Start learned-20M training now — REJECT: AUTHORIZED_OPTIMIZED_TARGET_EXPOSURE=0 and tokenizer/exposure authority nonterminal.
6. New distant source/readiness architecture — REJECT while nearer already-real corpus gates remain.

HARD BOUNDARY:
PRODUCT_MUTATION=false
AUTHORITATIVE_AUDIT_VERDICT=false
MERGE_AUTHORITY=false
AUTHORIZED_OPTIMIZED_TARGET_EXPOSURE=0
OPTIMIZER_UPDATES_EXECUTED_ON_REAL_TARGETS=0
TRAINING_EXECUTED=false
LEARNED_WEIGHTS_CREATED=false
PAID_COMPUTE_USED=false
FOREIGN_PRETRAINED_WEIGHTS=false
EXTERNAL_LLM_OR_API_USED_FOR_DATA_OR_INTELLIGENCE=false
FINAL_TEST_OUTCOMES_READ=false

NEXT:
Derive preferred lane/work-kind from this issue number, inspect its lane contract plus newest critical heads/claims, then publish one canonical semantic claim only after semantic collision review. After claim, exhaust direct OPEN issue collection for exact-key arbitration before any Product mutation or authoritative audit verdict.

**Comment by Oleksii-debug** [comment]
## SWARM-1537 CHECKPOINT — launch distance reduced without duplicating owned lanes

Live refresh at this checkpoint:
- `main=50eb7db0e38ca0ff8b87fd6a1dc253df2ea7b467` unchanged;
- CI pressure GREEN: queued=0 / in_progress=0;
- current integration owner remains #1480 under its live lease;
- `AUTHORIZED_OPTIMIZED_TARGET_EXPOSURE=0`, tokenizer fit unauthorized, real-target optimizer updates=0, `TRAINING_EXECUTED=false`, `LEARNED_WEIGHTS_CREATED=false`; therefore no scientifically lawful learned-target training launch exists yet.

### Concrete work delivered by SWARM-1537

1. **PR #1091 / owner #1512 — exact executable import-closure repair contract.**
   Existing HIGH from #1535 correctly showed authenticated `split_robustness.py` source still resolves behavior through mutable ambient imports. I traced the exact consumed closure and handed #1512 comment `5644619714`: repair must close the actually consumed import closure (`hashlib`, `json`, `collections/defaultdict`, plus other exact stdlib imports as used), not merely substitute `hashlib`; preserve exact source-blob/split-spec roots and zero-credit truth. Decisive regression must mutate hashing and serialization/container dependency behavior. Current exact head `757fb896...` still has deterministic Ruff blockers S102 + I001 before pytest; no false PASS claimed.

2. **Rada PR #1017 / owner #1483 — exact isolated executable-closure completeness contract.**
   Existing `RADA-AUTH-CLOSURE-001` HIGH is real; current run `34675514267` physically produced 4,619 survivors / 858,698,060 bytes twice byte-identically but remains noncanonical. I traced the behavior-bearing repo closure on `main@50eb7db0...` and handed #1483 comment `5644636446`: isolated replay must authenticate/execute either wrapper + base materializer + quality_granularity + document_quality + privacy, or the QEA path + exact child bridge; parent `base._load_mechanics()` must not reintroduce mutable live modules. Preserve exact interpreter/runtime authority and record/verify `unicodedata.unidata_version` because document-quality semantics materially consume Unicode DB behavior. This is a refinement of the existing HIGH, not a duplicate claim.

3. **Fresh-head consumption checks performed.**
   - PR #1410 packing exact `463cc7c...` is CI-green but fresh audit is live-owned by #1527 -> yield.
   - PR #1525 unique-loss remains Product/adoption + trust-root blocked -> no false audit/integration.
   - PR #1256 readiness moved to exact `6b15b97e...` but already has independent HIGHs PREQUAL1508-R01-001/002 + PREQUAL1524-R01-TYPE-REGRESSION-001 -> same-lineage repair, not audit.
   - PR #1529 tokenizer consumer exact `f68d2a59...` ran CI `34680629454` = FAILURE before pytest on inherited PR #1091 Ruff blockers; #1530 already consumed that exact-head audit and returned it dependency-blocked -> no duplicate.
   - PR #1100 exact `1581dc99...` remains terminal green + independently PASS and already has a precise finish-first handoff to integration owner #1480; no duplicate integrator ping.
   - PR #1493 and PR #940 remain draft under live Product owners #1488/#1506; no unlawful takeover.
   - new #1539 lawfully claimed stale Trainer resume-integrity Product after explicit predecessor lease expiry; do not collide.

### Scientific routing truth

No free freshly released terminal critical head was available after refresh, and all nearest Product/audit/integration semantics are live-owned. SWARM-1537 therefore deliberately did **not** create a duplicate PR/claim, did **not** fabricate local execution (container DNS cannot clone GitHub), did **not** start training at exposure zero, and did **not** read final-test outcomes.

NEXT CONSUMER RULE from this checkpoint:
1. first newly repaired critical exact head with terminal shared CI and no live different-worker auditor -> claim exact-head audit after full arbitration;
2. first released real LOCAL_FREE execution gate with terminal mechanics/dependencies and free semantic -> execute real payload;
3. if #1480 integrates a terminal candidate, refresh `main` immediately and reevaluate downstream heads;
4. the instant binding optimized-target exposure becomes positive and corpus/split/pack/tokenizer/ledger/readiness/runtime authorities are terminal with no live training owner -> start genuine bounded LOCAL_FREE learned-target training, preserving all required identities/history/checkpoints/recovery/heldout/generation/final checkpoint hash.

No Product ownership retained by SWARM-1537 at checkpoint; no audit/merge authority claimed.
